### PR TITLE
Migration guide for PrimaryScrollController Updates

### PR DESCRIPTION
**Pull request by Piinks**
Adds a migration guide for changes from https://github.com/flutter/flutter/pull/102099

## Presubmit checklist
- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/master/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

**Review by domesticmouse — APPROVED** [review]

File: src/release/breaking-changes/primary-scroll-controller-desktop.md
Comment (sfshaza2) [inline src/release/breaking-changes/primary-scroll-controller-desktop.md:53]:
```suggestion
will look up the `PrimaryScrollController` and instead call `shouldInherit` to
```

File: src/release/breaking-changes/primary-scroll-controller-desktop.md
Comment (sfshaza2) [inline src/release/breaking-changes/primary-scroll-controller-desktop.md:57]:
```suggestion
`automaticallyInheritForPlatforms` and `scrollDirection`, are evaluated in
```

File: src/release/breaking-changes/primary-scroll-controller-desktop.md
Comment (sfshaza2) [inline src/release/breaking-changes/primary-scroll-controller-desktop.md:63]:
```suggestion
`PrimaryScrollController.shouldInherit` returns true for vertical
`ScrollView`s. On desktop, this returns false by default.
```

File: src/release/breaking-changes/primary-scroll-controller-desktop.md
Comment (sfshaza2) [inline src/release/breaking-changes/primary-scroll-controller-desktop.md:87]:
```suggestion
Widgets that use the `PrimaryScrollController`, such as `NestedScrollView`,
`Scrollbar`, and `DropdownMenuButton` will experience no change to existing
functionality. Features like the iOS scroll-to-top will also continue to work as
expected without any migration.
```

File: src/release/breaking-changes/primary-scroll-controller-desktop.md
Comment (sfshaza2) [inline src/release/breaking-changes/primary-scroll-controller-desktop.md:93]:
```suggestion
side-by-side is common on desktop platforms, it isn't possible for
```

File: src/release/breaking-changes/primary-scroll-controller-desktop.md
Comment (sfshaza2) [inline src/release/breaking-changes/primary-scroll-controller-desktop.md:101]:
```suggestion
If more than one `ScrollView` was present previous to this change, the same
assertion (`ScrollController attached to multiple ScrollViews.`) would be thrown.
Now, on desktop platforms, users need to specify `primary: true` to
designate which `ScrollView` is the fallback to receive unhandled keyboard
`Shortcuts`.
```

**Review by sfshaza2 — COMMENTED** [review]
A few edits, but nothing huge.  :)

**Comment by sfshaza2** [comment]
Also, do you have a version where this was merged?

**Comment by Piinks** [comment]
> do you have a version where this was merged?

It was merged a while ago, but it does not have a build tag yet.

**Comment by domesticmouse** [comment]
> > do you have a version where this was merged?
> 
> It was merged a while ago, but it does not have a build tag yet.

@Piinks can you mark your PRs as Draft until you are ready for us to review and land them?

**Comment by Piinks** [comment]
> can you mark your PRs as Draft until you are ready for us to review and land them?

Why would it be a draft? This change has landed and customers inside of google will need this before it gets a version tag.

> do we have the "landed in version" info?

No. As stated above, it does not have a tag yet.

We used to generate version tags when Flutter rolled into Google, this was reflected in the dev channel. The dev channel is no longer being used, and we only tag versions when we roll to beta. This only happens ~ once a month, but in the meantime, Google customers have this untagged version. Should they not have a migration guide while we wait for a beta release?

Also, other migration guides have been merged for changes that have not even landed yet - so why would this not be ready? This guide does not have a version tag, and the change has not even landed yet: 

- https://github.com/flutter/flutter/pull/97972
- https://github.com/flutter/website/pull/7243/files
- https://docs.flutter.dev/release/breaking-changes/toggleable-active-color

Can you help me understand why that can land but this can't?

**Comment by sfshaza2** [comment]
OK, thanks for clarifying. I didn't realize that this was the result of retiring the dev branch.

**Pull request merged**
